### PR TITLE
Proposal: Add support for multiple cells types

### PR DESCRIPTION
**Pull request by syatyo**
# Proposal
Add support for multiple cells types

### CODE REVIEW

**Comment by rechsteiner** [comment]
Hi @syatyo! Thank you so much for your PR. I have a few comments on your proposal.

- I don't think we should add `cellReuseIdentifier` to the `PagingItem` protocol. Registering multiple cells is quite an edge-case and I don't think we should force everyone to implement it. There is always a one-to-one mapping between items and cells, so I think it's safe to use `String(describing: self)` of the item type as the cell identifier.

- Instead of changing `menuItemSources` to an array I was thinking of getting rid of that property altogether. I was thinking of making the API similar to the `UICollectionView` cell registration, like this: 

```Swift
pagingViewController.register(LoadingCell.self, for: LoadingItem.self)
pagingViewController.register(nib, for: LoadingItem.self)
```

 By default, `PagingTitleCell` will be registered. 

What do you think about that?

**Comment by syatyo** [comment]
Thank you for your comments @rechsteiner !

- It's good idea using `String(describing: self)`. I had wondering about using proper cell-reuseIdentifier in making a collection view cell. Your approach seems better 👏 

- New APIs are so simple and good!

I'll try to commit these changes.

**Comment by syatyo** [comment]
Hi @rechsteiner !

I have just committed them following your comments. Please review them🙇

File: Parchment/Classes/PagingViewController.swift [hunk 1]
```diff
@@ -330,6 +333,20 @@ open class PagingViewController:
     }, completion: nil)
   }
   
+  /// Register cell class for paging cell
+  /// - Parameter cellClass: paging cell's class
+  /// - Parameter pagingItemType: paging item type for specifying cell identifier
+  public func register(_ cellClass: AnyClass?, forCellWithPagingItemType pagingItemType: PagingItem.Type) {
```
Comment (syatyo) [inline Parchment/Classes/PagingViewController.swift]: I've added new apis as 'register(_ cellClass: AnyClass?, forCellWithPagingItemType pagingItemType: PagingItem.Type)'. I tried to use similar argument name to UIKit

File: Parchment/Classes/PagingController.swift [hunk 2]
```diff
@@ -653,10 +639,10 @@ extension PagingController: UICollectionViewDataSource {
   // MARK: UICollectionViewDataSource
   
   func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+    let pagingItem = visibleItems.items[indexPath.item]
     let cell = collectionView.dequeueReusableCell(
-      withReuseIdentifier: PagingController.CellIdentifier,
+      withReuseIdentifier: String(describing: type(of: pagingItem)),
```
Comment (syatyo) [inline Parchment/Classes/PagingController.swift]: Use `Type` of pagingItem instance to decide proper cell identifier.

File: Parchment/Classes/PagingOptions.swift [hunk 3]
```diff
@@ -5,11 +5,7 @@ public struct PagingOptions {
   /// The size for each of the menu items. _Default:
   /// .sizeToFit(minWidth: 150, height: 40)_
   public var menuItemSize: PagingMenuItemSize
-  
-  /// The class type for the menu item. Override this if you want
-  /// your own custom menu items. _Default: PagingTitleCell.self_
-  public var menuItemSource: PagingMenuItemSource
```
Comment (syatyo) [inline Parchment/Classes/PagingOptions.swift]: I get rid of old api

Comment (syatyo) [inline Parchment/Classes/PagingOptions.swift]: I get rid of old API.

File: Parchment/Classes/PagingController.swift (hunk 2 above)
Comment (syatyo) [inline Parchment/Classes/PagingController.swift]: I use `Type` name of pagingItem instance to decide proper cell identifier.

File: Parchment/Classes/PagingViewController.swift (hunk 1 above)
Comment (syatyo) [inline Parchment/Classes/PagingViewController.swift]: I've added a new API as register(_ cellClass: AnyClass?, forCellWithPagingItemType pagingItemType: PagingItem.Type). I tried to use similar argument name to UIKit.

Comment (rechsteiner) [reply Parchment/Classes/PagingViewController.swift]: Can you rename `forCellWithPagingItemType` to just `for`?

Comment (syatyo) [reply Parchment/Classes/PagingViewController.swift]: I completed to rename them!

Comment (rechsteiner) [reply Parchment/Classes/PagingViewController.swift]: Great! One more thing, any reason `AnyClass`/`UINib` is optional?

Comment (syatyo) [reply Parchment/Classes/PagingViewController.swift]: It's only for consistency to UIKit API.

**Comment by rechsteiner** [comment]
Awesome, thank you so much for adding this 🙌

**Pull request merged**